### PR TITLE
gitrepo: clone and open as non-bare

### PR DESCRIPTION
**Pull request by kopiczko**
Towards https://github.com/giantswarm/giantswarm/issues/7491

architect uses non-bare repositories.

### CODE REVIEW

File: pkg/gitrepo/repo.go [hunk 1]
```diff
@@ -48,13 +51,14 @@ func New(config Config) (*Repo, error) {
 		}
 	}
 
-	fs := osfs.New(config.Dir)
-	storage := filesystem.NewStorageWithOptions(fs, cache.NewObjectLRUDefault(), filesystem.Options{ExclusiveAccess: true})
+	worktree := osfs.New(config.Dir)
+	fs := osfs.New(filepath.Join(config.Dir, ".git"))
+	storage := filesystem.NewStorageWithOptions(fs, cache.NewObjectLRUDefault(), filesystem.Options{})
```
Comment (MarcelMue) [inline pkg/gitrepo/repo.go]: Why no longer exclusive access?

Comment (kopiczko) [reply pkg/gitrepo/repo.go]: Just in case. I thought it's premature optimization.

**Comment by kopiczko** [comment]
Tested with opsctl.

**Comment by kopiczko** [comment]
Tested with architect.

**Comment by kopiczko** [comment]
I'm going with that.

**Pull request merged**